### ISSUE TITLE
Links fail when domain name is included in link

### ISSUE THREAD

**Issue by stefvhuynh** [gatsbyjs/gatsby#3546]
When you have an anchor tag in markdown, links are not prefixed (https://github.com/gatsbyjs/gatsby/issues/3316); therefore opening the link in a new tab fails. To get around this, you can pass in the absolute url instead of just a relative url. 

However, when you just click on the link regularly, it will append the entire absolute url to the existing url. The reason is because of this line: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-catch-links/src/catch-links.js#L61. The entire `href` attribute gets passed in when the `host` of the `href` matches the `host` of the page you are on. 

Would it be possible to change this line to
```
cb(anchor.pathname)
```

**Comment by KyleAMathews** [comment]
Does the pathname exist on the anchor? If it does that sounds fine.

**Comment by stefvhuynh** [comment]
yeah, usually for these links, you'd pass in `<a href="/some/path">`. so it's working now because `anchor.getAttribute('href')` returns that property verbatim. but it won't work if you do `<a href="https://mydomain.com/some/path">`. This will result in pushing `mydomain.com/some/path` to the end of the current url. However, if you use `anchor.pathname`, you will get `/some/path` in both examples, which will result in correct navigation behavior.

**Comment by KyleAMathews** [comment]
We should detect if the link is pointed to a different domain and then let the browser handle the link click normally.

A PR for this would be great!

**Comment by alem0lars** [comment]
Any updates?

**Comment by kakadiadarpan** [comment]
@alem0lars There's a PR already open for this issue - #7779

Once it is merged the fix will be available for Gatsby v1, this issue has already been fixed for Gatsby v2. If you want to give that a try, we have nicely written [migration guide](https://www.gatsbyjs.org/docs/migrating-from-v1-to-v2/).
